src/lib/response-triggers/finish.ts

```diff
@@ -35,33 +39,31 @@ export default class Finish extends Trigger {
 	}
 
 	async createMessage() : Promise<MessageObject> {
-		let winnersPromise = this.winners.createMessage();
-		let resultsPromise = this.results.createMessage();
-		__logger.debug('Finish promises created');
+		let winnersMarkdown: string;
+		let resultsMarkdown: string;
 
-		return Promise.all([winnersPromise, resultsPromise])
-		.then((values : MessageObject[]) => {
-			__logger.information('Winners and Results promises executed');
-			return this.reset.createMessage()
-			.then((data) => {
-				__logger.information('Reset promise executed');
-				let message = `## Winners\n\n` + values[0].markdown + '\n\n';
-				message += values[1].markdown;
-				message += '\n\n' + data.markdown;
-				return {
-					markdown: message
-				};
-			}).catch((error) => {
-				__logger.error(`Error clearing pegs:\n${error.message}`);
-				return {
-					markdown: `error clearing pegs`
-				};
-			});
-		}).catch((error) => {
+		try {
+			winnersMarkdown = await this.winnersService.returnWinnersResponse();
+			resultsMarkdown = await this.resultsService.returnResultsMarkdown();
+		} catch(error) {
 			__logger.error(`Error returning winners or results:\n${error.message}`);
```

Comment:
Oh boy... this is gonna make it really hard to merge my logging improvement branch.

Might have to just scrap it and start again after this is done...

Reply:
I apologise haha